squire_lib/src/players/player_registry.rs

```diff
@@ -96,21 +102,26 @@ impl PlayerRegistry {
             }
             None => {
                 // Not re-registering
-                match self.name_and_id.contains_left(&account.user_name) {
-                    true => Err(PlayerAlreadyRegistered),
-                    false => {
-                        let name = account.get_user_name();
-                        let plyr = Player::from_account(account);
-                        let digest = Ok(plyr.id);
-                        self.name_and_id.insert(name, plyr.id);
-                        self.players.insert(plyr.id, plyr);
-                        digest
-                    }
-                }
+                let Some(name) = (!self.name_and_id.contains_left(&account.user_name))
+                    .then_some(account.get_user_name())
+                    .or(tourn_name)
```

Comment:
If we get to this point, we should also check that the `tourn_name` is also not already in use.